### PR TITLE
Add up and down arrow keys handling (resolves #7180)

### LINKED ISSUE #7180: TGCommandLinePlugin Up-Down Arrows and Tab Keys
### Explain what you would like to see improved
The TGCommandLinePlugin is great for external CMake/C++ applications that use some ROOT classes / TApplication and GUI classes TGFrame, where you want to have some interactive debugging options via Cling. To make it even more user-friendly, it would be nice if the following things were improved:
- Pressing Up and Down Arrow Keys changes the selected command in the ComboBox.
- Pressing Tab Key propagates to the command line to give command suggestions / autocompletion, as it happens in the real ROOT terminal
- Typed commands are added to the ROOT history file, so that the next time 

### Optional: share how it could be improved
Changes in TGTextEntry:
```
    virtual  void        UpDownArrowsPressed(Int_t);                           //*SIGNAL*

case kKey_Down:
          CursorOutDown();
          UpDownArrowsPressed(1);
          break;
case kKey_Up:
          CursorOutUp();
          UpDownArrowsPressed(0);
          break;
```

Changes in TGCommandLinePlugin
```
fCommand->Connect("TabPressed()", "TGCommandPlugin", this,
                      "HandleTab()");
fCommand->Connect("UpDownArrowsPressed()", "TGCommandPlugin", this,
                      "HandleArrows()");

void TGCommandPlugin::HandleTab()
 {
  //redirect CLI suggestions back to terminal
}


void TGCommandPlugin::HandleArrows()
 {
  //Scroll one up or down in combobox
}

if (app->InheritsFrom("TRint") || fForceFlag ) // accept also if a user set an fForceFlat explicitly?
          Gl_histadd((char *)string);
```

### To Reproduce
- Open tutorials/gui/guiWithCINT.C
- Try to use up and down arrow keys to scroll between line commands

### Setup
1. ROOT 6.23/01
2. Linux
3. Self-built

### PR DESCRIPTION
**Pull request by bellenot**

### CODE REVIEW

File: gui/gui/src/TGCommandPlugin.cxx [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// TGCommandPlugin                                                      //
+//                                                                      //
+// Class used to redirect command line input/output.                    //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
```
Comment (couet) [inline gui/gui/src/TGCommandPlugin.cxx:18]: Why not using the doxygen format:

/** \class TGCommandPlugin
Class used to redirect command line input/output.
**/

The comment will show here: https://root.cern/doc/master/classTGCommandPlugin.html

File: gui/gui/src/TGCommandPlugin.cxx [hunk 2]
```diff
@@ -174,3 +263,19 @@ Bool_t TGCommandPlugin::HandleTimer(TTimer *t)
...
+////////////////////////////////////////////////////////////////////////////////
+/// The function SetHistAdd() is needed for a standalone TApplication to log the
+/// TGCommandPlugin commands into a ROOT history file.
+/// However, this function has no effect if the user does not explictly set on
+/// his standalone application the name of the ROOT history file.
+/// To log into the default ROOT history file, call this on the user-side of the
+/// code:
+///    Gl_histinit(gEnv->GetValue("Rint.History", gSystem->HomeDirectory()));
+/// Otherwise, replace the argument of Gl_histinit with a text file name you want
+/// to use for application-specific logging.
```
Comment (couet) [inline gui/gui/src/TGCommandPlugin.cxx:276]: here it is already in doxygen format ! Good ! :-)

**Review by couet — COMMENTED** [review]
Doxygen formatting could be used for the header.

File: gui/gui/src/TGCommandPlugin.cxx (hunk 1 above)
Comment (bellenot) [reply gui/gui/src/TGCommandPlugin.cxx:18]: OK, will change it, thanks!

Comment (couet) [reply gui/gui/src/TGCommandPlugin.cxx:18]: Then the other GUI classes might need a similar change. 
But that can be done step by step.

**Review by couet — APPROVED** [review]

File: gui/gui/src/TGCommandPlugin.cxx (hunk 1 above)
Comment (bellenot) [reply gui/gui/src/TGCommandPlugin.cxx]: Yes, it's what I saw... Same for graf2d 😉

**Comment by bellenot** [comment]
Failures are unrelated. Merging.

**Pull request merged**